TestFoundation/TestProcess.swift

```diff
@@ -505,6 +512,68 @@ class TestProcess : XCTestCase {
         XCTAssertTrue(helper.process.resume())
     }
 
+
+    func test_redirect_stdin_using_null() {
+        let url = URL(fileURLWithPath: "/bin/cat", isDirectory: false)
+        let task = Process()
+        task.executableURL = url
+        task.standardInput = FileHandle.nullDevice
+        XCTAssertNoThrow(try task.run())
+        task.waitUntilExit()
+    }
+
+
+    func test_redirect_stdout_using_null() {
+        let url = URL(fileURLWithPath: "/usr/bin/env", isDirectory: false)
+        let task = Process()
+        task.executableURL = url
+        task.standardOutput = FileHandle.nullDevice
+        XCTAssertNoThrow(try task.run())
+        task.waitUntilExit()
+    }
+
+    func test_redirect_stdin_stdout_using_null() {
+        let url = URL(fileURLWithPath: "/bin/cat", isDirectory: false)
+        let task = Process()
+        task.executableURL = url
+        task.standardInput = FileHandle.nullDevice
+        task.standardOutput = FileHandle.nullDevice
+        XCTAssertNoThrow(try task.run())
+        task.waitUntilExit()
+    }
+
+
+    func test_redirect_stderr_using_null() throws {
+        let url = URL(fileURLWithPath: "/usr/bin/env", isDirectory: false)
+        let task = Process()
+        task.executableURL = url
+        task.standardError = FileHandle.nullDevice
+        XCTAssertNoThrow(try task.run())
+        task.waitUntilExit()
+    }
+
+
+    func test_redirect_all_using_null() throws {
+        let url = URL(fileURLWithPath: "/bin/cat", isDirectory: false)
```

Comment:
None of these tests will run outside of a UNIX system. Can we run xdgTestHelper instead, since it's the only executable whose presence we rely on? (Perhaps with a no-op flag.)

Reply:
I removed all of the unix specific commands (`/bin/cat`, `/bin/env` etc) and added simple versions of them as options in `xdgTestHelper`.